CMakeLists.txt

```diff
@@ -56,10 +55,10 @@ endif()
 
 list(APPEND zoslib_cflags -Wno-parentheses -Wno-unused-value)
 
-if(${CMAKE_C_COMPILER} MATCHES ibm-clang64)
+if(${CMAKE_C_COMPILER} MATCHES ibm-clang64 OR ${CMAKE_C_COMPILER} MATCHES clang)
   list(APPEND zoslib_cflags
     -fgnu-keywords
-    -march=arch10
+    -march=arch14
```

Comment:
We went with ARCH10 because the ALS for V2R4 is arch(10). 

Gaby made some changes in the main branch for genentropy if this is related to that: https://github.com/ibmruntimes/zoslib/blob/main/src/zos-getentropy.cc

Which reminds me, we should probably merge zopen with main soon before we diverge further.

Reply:
I realized that we had used a clang wrapper to process the pragma options in Node.js, so that probably won't work here.

Reply:
That's why I changed it to 14.  I mentioned to CW, that hack using pragma option does result in the compiler issuing a warning for unsupported option.  Rather than a PP directive, I suggest using a comment, eg. `//pragma options("....")` and even change options to something more related to your tool.  

We should remove this line from zos-getentropy.cc since that tool isn't used in the build of this source.

Reply:
as a side point, we don't support a -mtune option on z/os.  We should remove all usage from all of the ports that support clang.

Reply:
I just tried this with xlclang and it looks like the new check is also true for xlclang since xlclang contains clang.

Reply:
I solved this by switching the true/false and checking for xlclang.

Reply:
In our internal repo we changed it to use [`STREQUAL`](https://cmake.org/cmake/help/latest/command/if.html#strequal):

```cmake
get_filename_component(COMP ${CMAKE_C_COMPILER} NAME CACHE)

if(${COMP} STREQUAL clang OR ${COMP} STREQUAL ibm-clang64)
```

`get_filename_component()` was needed to extract just the filename because `CMAKE_C_COMPILER` is the full path.